### PR TITLE
[Release 1.1] Upgrade MSAL version to latest

### PR DESCRIPTION
**Pull request by cheenamalhotra**

### CODE REVIEW

File: src/Microsoft.Data.SqlClient/netcore/src/Microsoft.Data.SqlClient.csproj [hunk 1]
```diff
@@ -601,7 +601,7 @@
     <PackageReference Include="System.Diagnostics.DiagnosticSource" Version="4.5.0" />
     <PackageReference Include="System.Threading.Tasks" Version="4.3.0" />
     <PackageReference Include="System.Memory" Version="4.5.1" />
-    <PackageReference Include="Microsoft.Identity.Client" Version="3.0.8" />
+    <PackageReference Include="Microsoft.Identity.Client" Version="4.7.1" />
```
Comment (JRahnama) [inline src/Microsoft.Data.SqlClient/netcore/src/Microsoft.Data.SqlClient.csproj:604]: Should not we add the version to version props?

File: src/Microsoft.Data.SqlClient/netfx/src/Microsoft.Data.SqlClient.csproj [hunk 2]
```diff
@@ -349,7 +349,7 @@
       <Version>1.1.*</Version>
     </PackageReference>
     <PackageReference Include="Microsoft.Identity.Client">
-      <Version>3.0.8</Version>
+      <Version>4.7.1</Version>
```
Comment (JRahnama) [inline src/Microsoft.Data.SqlClient/netfx/src/Microsoft.Data.SqlClient.csproj:352]: Same here about version props?

File: src/Microsoft.Data.SqlClient/netcore/src/Microsoft.Data.SqlClient.csproj (hunk 1 above)
Comment (cheenamalhotra) [reply src/Microsoft.Data.SqlClient/netcore/src/Microsoft.Data.SqlClient.csproj:604]: Not here since this is old branch.
Optimization was done much later.

Comment (JRahnama) [reply src/Microsoft.Data.SqlClient/netcore/src/Microsoft.Data.SqlClient.csproj:604]: Thanks for clarification. I missed the branch.

**Comment by cheenamalhotra** [comment]
Discarding the PR and upgrade no longer needed.

**Pull request closed without merging**